src/sf45b.cpp

```diff
@@ -1,219 +1,239 @@
+    ROS_INFO("Firmware: %.16s (%d)", firmwareVersionStr, firmwareVersion);
+    ROS_INFO("Serial: %.16s", serialNumber);
+
+    return 0;
+}
 
-	if (Params->lowAngleLimit < -160) Params->lowAngleLimit = -160;
-	else if (Params->lowAngleLimit > -10) Params->lowAngleLimit = -10;
+int driverScanStart(lwSerialPort *Serial, lwSf45Params *Params) {
+    if (!lwnxCmdWriteUInt32(Serial, 27, 0x101)) {
+        return 1;
+    }
 
-	if (Params->highAngleLimit < 10) Params->highAngleLimit = 10;
-	else if (Params->highAngleLimit > 160) Params->highAngleLimit = 160;
-}
+    if (!lwnxCmdWriteUInt8(Serial, 66, Params->updateRate)) {
+        return 1;
+    }
 
-int driverStart(lwSerialPort** Serial, const char* PortName, int32_t BaudRate) {
-	platformInit();
+    if (!lwnxCmdWriteUInt16(Serial, 85, Params->cycleDelay)) {
+        return 1;
+    }
 
-	lwSerialPort* serial = platformCreateSerialPort();
-	*Serial = serial;
-	if (!serial->connect(PortName, BaudRate)) {
-		ROS_ERROR("Could not establish serial connection on %s", PortName);
-		return 1;
-	};
+    if (!lwnxCmdWriteFloat(Serial, 98, Params->lowAngleLimit)) {
+        return 1;
+    }
 
-	// Disable streaming of point data. (Command 30: Stream)
-	if (!lwnxCmdWriteUInt32(serial, 30, 0)) { return 1; }
+    if (!lwnxCmdWriteFloat(Serial, 99, Params->highAngleLimit)) {
+        return 1;
+    }
 
-	// Read the product name. (Command 0: Product name)
-	char modelName[16];
-	if (!lwnxCmdReadString(serial, 0, modelName)) { return 1; }
+    if (!lwnxCmdWriteUInt32(Serial, 30, 5)) {
+        return 1;
+    }
 
-	// Read the hardware version. (Command 1: Hardware version)
-	uint32_t hardwareVersion;
-	if (!lwnxCmdReadUInt32(serial, 1, &hardwareVersion)) { return 1; }
+    return 0;
+}
 
-	// Read the firmware version. (Command 2: Firmware version)
-	uint32_t firmwareVersion;	
-	if (!lwnxCmdReadUInt32(serial, 2, &firmwareVersion)) { return 1; }
-	char firmwareVersionStr[16];
-	lwnxConvertFirmwareVersionToStr(firmwareVersion, firmwareVersionStr);
+struct lwDistanceResult {
+    float x;
+    float y;
+    float z;
+};
+
+int driverScan(lwSerialPort *Serial, lwDistanceResult *DistanceResult) {
+    lwResponsePacket response;
 
-	// Read the serial number. (Command 3: Serial number)
-	char serialNumber[16];
-	if (!lwnxCmdReadString(serial, 3, serialNumber)) { return 1; }
+    if (lwnxRecvPacket(Serial, 44, &response, 1000)) {
+        int16_t distanceCm = (response.data[5] << 8) | response.data[4];
+        int16_t angleHundredths = (response.data[7] << 8) | response.data[6];
 
-	ROS_INFO("Model: %.16s", modelName);
-	ROS_INFO("Hardware: %d", hardwareVersion);
-	ROS_INFO("Firmware: %.16s (%d)", firmwareVersionStr, firmwareVersion);
-	ROS_INFO("Serial: %.16s", serialNumber);
+        float distance = distanceCm / 100.0f;
+        float angle = angleHundredths / 100.0f;
+        float faceAngle = (angle - 90) * M_PI / 180.0;
 
-	return 0;
+        DistanceResult->x = distance * -cos(faceAngle);
```

Comment:
Careful with these orientation changes, is this intended? Same question faceAngle

Reply:
This logic has not been changed